vertical-pod-autoscaler/pkg/recommender/input/controller_fetcher/controller_fetcher.go

```diff
@@ -248,6 +284,15 @@ func (f *controllerFetcher) isWellKnown(key *ControllerKeyWithAPIVersion) bool {
 	return exists
 }
 
+func (f *controllerFetcher) getScaleForResource(namespace string, groupResource schema.GroupResource, name string) (controller *autoscalingapi.Scale, err error) {
+	if ok, scale, err := f.controllerCache.Get(namespace, groupResource, name); ok {
+		return scale, err
+	}
+	scale, err := f.scaleNamespacer.Scales(namespace).Get(context.TODO(), groupResource, name, metav1.GetOptions{})
```

Comment:
I agree that it would be nicer to use if cache took care of fetching entries. But I think it would make code more complicated. Since this is the only place I expect to use the cache I think it's overall better to keep things simpler.

I named it cache *storage* to emphasise that it's doing less than one might hope.

Reply:
That makes sense, I'm not gonna be supper stubborn about it :)

Reply:
Actually it might be useful to rename the field as well, so it's clear at call site it's not a full cache but rather a storage.